folio-token/src/main/java/org/folio/spring/config/properties/FolioEnvironment.java

```diff
@@ -0,0 +1,38 @@
+package org.folio.spring.config.properties;
+
+import static java.lang.System.getProperty;
+import static java.lang.System.getenv;
+import static org.apache.commons.lang3.StringUtils.firstNonBlank;
+
+import jakarta.validation.constraints.NotEmpty;
+import jakarta.validation.constraints.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.validation.annotation.Validated;
+
+@Data
+@Validated
+@Configuration
+@NoArgsConstructor
+@AllArgsConstructor(staticName = "of")
+@ConfigurationProperties(prefix = "folio")
+public class FolioEnvironment {
+
+  @NotEmpty
+  @Pattern(regexp = "[a-zA-Z0-9\\-_]+", message = "Value must follow the pattern: '[\\w0-9\\-_]+'")
+  private String environment;
+
+  private String okapiUrl;
+
+  /**
+   * Return folio env name from environment or system properties as {@link String} object.
+   *
+   * @return folio env name.
+   */
+  public static String getFolioEnvName() {
+    return firstNonBlank(getenv("ENV"), getProperty("env"), getProperty("environment"), "folio");
+  }
+}
```

Comment:
Where this class is used?

Reply:
in SystemUserService
![image](https://github.com/folio-org/folio-spring-support/assets/104345088/42b446ab-232e-4058-a1a3-f19b22d94851)